src/OpenTracing/ISpan.cs

```diff
@@ -1,164 +1,177 @@
-        /// <param name="value">The value to be stored.</param>
-        /// <returns>The current <see cref="ISpan"/> instance for chaining.</returns>
         ISpan SetTag(string key, int value);
 
         /// <summary>
-        /// Adds a tag to the span.
+        ///     Same as <see cref="SetTag(string,string)" /> but for numeric values.
         /// </summary>
-        /// <param name="key">If there is a pre-existing tag set for <paramref name="key"/>, it is overwritten.</param>
-        /// <param name="value">The value to be stored.</param>
-        /// <returns>The current <see cref="ISpan"/> instance for chaining.</returns>
-        ISpan SetTag(string key, string value);
+        ISpan SetTag(string key, double value);
 
         /// <summary>
-        /// <para>Log key:value pairs to the span with the current timestamp.</para>
-        /// <para>CAUTIONARY NOTE: Not all Tracer implementations support key:value log fields end-to-end.
-        /// It is possible to pass a list of key:value pairs instead of a dictionary. However, the behavior
-        /// for lists with duplicate keys is undefined.
-        /// Caveat emptor.</para>
+        ///     Log key:value pairs to the Span with the current walltime.
+        ///     <para>
+        ///         <em>CAUTIONARY NOTE:</em> not all Tracer implementations support key:value log fields end-to-end.
+        ///         Caveat emptor.
+        ///     </para>
+        ///     <para>
+        ///         A contrived example:
+        ///         <code>
+        /// span.Log(
+        /// new Dictionary{string, object}
+        /// {
+        ///   { "event", "soft error" },
+        ///   { "type", "cache timeout" },
+        ///   { "waited.millis", 1500 },
+        /// });
+        /// </code>
+        ///     </para>
         /// </summary>
         /// <param name="fields">
-        ///   <para>key:value log fields. Tracer implementations should support string, numeric, and boolean values;
-        ///   some may also support arbitrary objects.</para>
-        ///   <para>The behavior for lists with duplicate keys is undefined.</para>
+        ///     key:value log fields. Tracer implementations should support String, numeric, and boolean values;
+        ///     some may also support arbitrary objects.
         /// </param>
-        /// <returns>The current <see cref="ISpan"/> instance for chaining.</returns>
-        /// <example>
-        /// <code>
-        /// span.Log(new Dictionary&lt;string, object&gt; {
-        ///     { "event", "soft error" },
-        ///     { "type", "cache timeout" },
-        ///     { "waited.millis", 1500 }
-        /// });
-        /// </code>
-        /// </example>
-        ISpan Log(IEnumerable<KeyValuePair<string, object>> fields);
+        /// <returns>The Span, for chaining</returns>
+        /// <seealso cref="Log(string)" />
+        ISpan Log(IDictionary<string, object> fields);
```

Comment:
Using IDictionary has quite a big performance hit and it no longer allows the usage of linked lists. IEnumerable<KeyValuePair...> allows for all scenarios and is also used in .NET's Activity type. The drawback is that it potentially allows duplicates. I'd like to have this discussed properly in a separate issue - but this could also happen after a merge of course.

Reply:
The concern is that you cannot index into IEnumerable (in fact demoting anything down to IEnumerable is usually a perf hit).

That said, this is the `Log` method, I'm not sure why I changed this one :) Unless there's reasonable assumption that an implementation may want a specific field and not want to iterate over everything. In that case, this is >> perf

RE: no longer allows the usage of linked lists, you can implement IDictionary with a LinkedList (doing so means finding a particular item is inefficient, but no more inefficient than doing so on LinkedList directly). It'd be possible to provide an extension method that can take IEnumerable<KeyValuePair> or, if implementors will really really want indexing into it to be performant then providing two methods, one that takes IEnumerable and one that takes IDictionary so that implementor can appropriately react to the functionality enabled by the input

Reply:
True, I should have written "Using the concrete `Dictionary`-type has quite a big performance hit" (since it does a uniqueness check on every addition). IDK - we should discuss this in #44

Reply:
Agreed, reverting all IDictionaries until #44 is resolved now that you're back around so that can get traction :)